### ISSUE TITLE
Copy/Paste Unix style...

### ISSUE THREAD

**Issue by liar666** [jupyterlab/jupyterlab#5318]
Hi,

I've read a number of past issues about enabling the copy/paste using the OS's clipboard and apparently the provided shortcuts work (Ctrl-C/V or Ctrl-Insert/Maj-Insert).

Unfortunately, I'm a Unix (Linux/*BSD) for many years now, and I cannot think of copy/pasting in any other way than selecting with the mouse (automatically copying to the clipboard) and clicking the middle button (paste).

Would that be possible to implement that in JupyterLab?

**Comment by martinRenou** [comment]
This would be a great feature in JupyterLab 

cc. @zerline

**Comment by jasongrout** [comment]
Does this not work right now (on unix?) as part of the browser/xwindows?

**Comment by martinRenou** [comment]
No, it works on the Github website, but not in JupyterLab. I also thought it would be there for free.

**Comment by jasongrout** [comment]
Does it work in the browser in general on unix? Does it not work everywhere in jupyterlab, or is there one specific place you are missing it?

**Comment by martinRenou** [comment]
So apparently it copies the text I select in JupyterLab, because I'm able to paste it in the browse bar. But I'm not able to paste the text in the JupyterLab interface (in a cell for example).

**Comment by jasongrout** [comment]
Oh, interesting. Sounds like the middle click doesn't work for some reason. Is that generating a contextmenu event or something that we are overriding?

**Comment by jasongrout** [comment]
You can try debugging the mouse click events, and see where this event is getting captured and overridden.

**Comment by dhirschfeld** [comment]
It would be great if this could work on Windows too! Possibly via a configurable.

**Comment by zerline** [comment]
My own experience on Linux is that you cannot copy (unix-style, just with the mouse) from a notebook and paste into another application  (e.g. into the terminal, text editor ..). To do that, you need to C-c in the notebook, then use the right menu in your application, which is really cumbersome.
NB This is not specific to the notebook, I think it comes from the browser itself.

**Comment by ianhi** [comment]
Things have changed somewhat, in jupyterlab 2.1.2 I find the following:

Chrome: You can both paste and copy as expected :)
Firefox:  You can paste with middle click, but cannot highlight to copy (even into the browser bar). Looks like this is being tracked in firefox: https://bugzilla.mozilla.org/show_bug.cgi?id=1593761


Also, middle clicking to close a different jupyter tab will paste text into an open editor on Firefox but not on Chrome (I can open this in a new issue if it deserves that):

![middle-click](https://user-images.githubusercontent.com/10111092/82136236-65cf4000-97d9-11ea-8bff-395c4ea6fd6f.gif)
This is probably related to: https://github.com/codemirror/CodeMirror/issues/4900#issuecomment-629743292 but in that issue it is both Chrome and Firefox, and this doesn't happen if you click above the editor, in contrast to this where it's only firefox and happens when I click above the editor.

**Comment by matrs** [comment]
@ianhi , the copy problem is still an issue for the notebook code cells (I think that's the problem here, not the text editor, as you're showing).
This is still a issue in 2022. Im using `jupyterlab 2.2.9` and `notebook 6.4.0` and can't copy content, which is in my clipbpoard and can copy with `Ctrl+c`.

**Comment by sorenwacker** [comment]
Pasting is not a problem, but I am unable to select and copy from the terminal.

**Comment by krassowski** [comment]
@sorenwacker can you share more details? Selecting and copying is supported and should work well (aside from browser limitations such as Firefox restricting some clipboard operations). Maybe worth opening a new issue?

https://github.com/user-attachments/assets/7a039c73-231b-4ff8-beac-afedcbc4a58b
